asyncpg/types.py

```diff
@@ -228,6 +228,28 @@ def as_string(self):
 
         return s.strip()
 
+    def to_int(self, bitorder):
```

Comment:
Is there a particular reason to make "bitorder" a thing?  At least, Postgres padding semantics seems to indicate that bitstrings have 'big' bitorder.  I would just go with this assumption.

Reply:
'little' is what I wanted for my case, for the typical little-endian advantage of giving consistent integer values for casts between Postgres bits of different lengths (and it gets even more annoying with varbits).

But yeah, I made the same assumption about Postgres padding / what people would expect from running `bin()` on the resulting integer, so I added bitorder by analogy to python's `int.to/from_bytes`. Potentially, unlike to/from_bytes, bitorder should have a default as 'big' here?

Reply:
> typical little-endian advantage of giving consistent integer values for casts between Postgres bits of different lengths

Can you elaborate on this please? It seems to me that "big" endians are actually more consistent given the right-padding (as done by `int.to_bytes(length=N)` and by `(CAST '111' AS bit(N))`

Reply:
With a little-endian bit-order, `B'101'::bit(10)` and `B'101'::bit(20)` result in the same integer. This is desirable if bits are being used to store collections of flags, because (for example) if a schema using bit(32) later needed to be migrated to bit(64) to support more flags, the values in Python would not need to change (assuming the usual C or Python enum.IntFlag conventions of assigning enum names to bits).

Reply:
If we go by the byte-order endianness definition where 'endian` actually means the *starting* bit, then 'little` is what we want here.

Reply:
Let's just assume the starting bit is least-significant, and omit the `bitorder` argument until someone comes along with a legitimate use case for it.

Reply:
Hmm. If we were going to only have one, it should be big-endian (despite the fact that that's not the one that personally want :cry:), because that's what Postgres currently uses when casting between ints and bits. Also, as you pointed out in the `from_int` case, inferring the length from the integer only makes sense when using big endian.

Reply:
Ah. I didn't realize the int casts were big-endian.  We can't have it nice here, it seems. Let's keep the argument then.

Please add docstrings to the new methods (with `.. versionadded:: 0.18.0`).